### PR TITLE
return correct page

### PR DESCRIPTION
**Pull request by laazy**
- fix bug in `draw_page`

### CODE REVIEW

File: fundbot/plugins/fund/search_impl.py [hunk 1]
```diff
@@ -25,8 +25,7 @@ def cache_add(search_args: list, result: str) -> str:
 
 def draw_page(arg_list):
     try:
-        arg_list.pop(len(arg_list) - 1)
-        return int(arg_list[-1]) - 1
+        return int(arg_list.pop(len(arg_list) - 1)) - 1
```
Comment (Veiasai) [inline fundbot/plugins/fund/search_impl.py:28]: does it not remove the last element default?
arg_list.pop() ??

Comment (laazy) [reply fundbot/plugins/fund/search_impl.py]: still have bug, but I've fixed it.
should pop after `int arg_list[-1]` succeeds.

File: fundbot/plugins/fund/search_impl.py [hunk 2]
```diff
@@ -25,10 +25,11 @@ def cache_add(search_args: list, result: str) -> str:
 
 def draw_page(arg_list):
     try:
-        arg_list.pop(len(arg_list) - 1)
-        return int(arg_list[-1]) - 1
+        page = int(arg_list[-1]) -1
+        arg_list.pop()
```
Comment (Veiasai) [inline fundbot/plugins/fund/search_impl.py:29]: it seems that we made it so hard to parse the command.
1. I don't think we should do pop here, the arg_list should be const.
2. I feel it's better to create a new list like,  search_args_list ? whatever it is. don't change the original list.
3. Also, I wonder if anybody want to search with number?  `#search 10086`,  the number is the filter, not page index,  is it needed?  I'd rather to implement a clear command:  `#searchpage 5 10086` ?

Comment (laazy) [reply fundbot/plugins/fund/search_impl.py:29]: Thanks for your comments. 
1.  I write a `draw_page` function because I don't want to write `try ... except` within a long function and I want to make a pipeline-like parsing. The word **draw** means that it will extract `page` from `arg_list.` 
2. This is a **bug-fix** pr, so I will not discuss the necessity of previous feature pr. We may refactor the code and implement this requirement in another pr.
3. If anyone want to search with number, just append 1 at last. eg: `#search 10086 1`.

Comment (Veiasai) [reply fundbot/plugins/fund/search_impl.py:29]: 1. sorry, I don't follow. I just suspect it's not a good idea to change the arg list. You can still use `draw_page`, and let it return a new list and page.
2. ok, approved.
3. ah, sounds like a good temporary solution, I prefer to improve it later on.

**Review by Veiasai — APPROVED** [review]
